test/priceUtils.js

```diff
@@ -1,5 +1,124 @@
+const BN = require("bn.js")
 const assert = require("assert")
-const { isPriceReasonable } = require("../scripts/utils/price-utils")(web3, artifacts)
+const { toErc20Units } = require("../scripts/utils/printing_tools")
+const { getOutputAmountFromPrice, getUnlimitedOrderAmounts, isPriceReasonable } = require("../scripts/utils/price-utils")(web3, artifacts)
+
+const max128 = new BN(2).pow(new BN(128)).subn(1)
+const floatTolerance = new BN(2).pow(new BN(52)) // same tolerance as float precision
+
+const assertEqualUpToFloatPrecision = function (value, expected) {
+  const differenceFromExpected = value.sub(expected).abs()
+  assert(differenceFromExpected.mul(floatTolerance).lt(expected))
+}
+
+describe("getOutputAmountFromPrice", () => {
+  it("computes the right output amount", () => {
+    const testCases = [
+      {
+        price: 160,
+        inputAmountString: "1",
+        inputDecimals: 18,
+        outputDecimals: 6,
+        expectedOutputAmountString: "160",
+      },
+      {
+        price: 1/160,
+        inputAmountString: "160",
+        inputDecimals: 6,
+        outputDecimals: 18,
+        expectedOutputAmountString: "1",
+      },
+      {
+        price: 0.000125,
+        inputAmountString: "8000",
+        inputDecimals: 8,
+        outputDecimals: 18,
+        expectedOutputAmountString: "1",
+      },
+      {
+        price: 10**30,
+        inputAmountString: "0.000000000000000000000001", // 10**-24
+        inputDecimals: 100,
+        outputDecimals: 1,
+        expectedOutputAmountString: "1000000",
+      },
+      {
+        price: 10.1,
+        inputAmountString: "1",
+        inputDecimals: 0,
+        outputDecimals: 70,
+        expectedOutputAmountString: "10.1",
+      },
+    ]
+    for (const { price, inputAmountString, inputDecimals, outputDecimals, expectedOutputAmountString } of testCases) {
+      const inputAmount = toErc20Units(inputAmountString, inputDecimals)
+      const expectedOutputAmount = toErc20Units(expectedOutputAmountString, outputDecimals)
+      const outputAmount = getOutputAmountFromPrice(price, inputAmount, inputDecimals, outputDecimals)
+      assertEqualUpToFloatPrecision(outputAmount, expectedOutputAmount)
```

Comment:
It's nice that you first multiply and then later divide by the inputDecimials  and outputDecimals, as this allows to easy test verification